Swiften/MIX/MIXRegistry.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#include <Swiften/MIX/MIXRegistry.h>
+#include <Swiften/Base/Log.h>
+
+namespace Swift {
+
+MIXRegistry::MIXRegistry(const JID& ownJID, IQRouter* iqRouter, XMPPRoster* xmppRoster) : ownJID_(ownJID), iqRouter_(iqRouter), xmppRoster_(xmppRoster) {
+    scopedConnection_ = xmppRoster_->onInitialRosterPopulated.connect(boost::bind(&MIXRegistry::syncRegistryWithRoster, this));
+}
+
+MIXRegistry::~MIXRegistry() {
+
+}
+
+void MIXRegistry::syncRegistryWithRoster() {
```

Comment:
This can be a C++11 lambda instead in the ctor, as the method name doesn't match project convention anyway (`handle*`).